app/src/org/commcare/utils/AndroidUtil.java

```diff
@@ -69,4 +76,34 @@ public static HashMap<String, String> bundleAsMap(Bundle bundle) {
     public static boolean isGooglePlayServicesAvailable(Context ctx) {
         return GoogleApiAvailability.getInstance().isGooglePlayServicesAvailable(ctx) == ConnectionResult.SUCCESS;
     }
+
+
+
+    /**
+     * Attaches a WindowInsetsListener to the root view of the activity for devices running Android 15+. This
+     * listener applies padding to the view based on the system window insets.
+     *
+     * @param activity   The activity to which the listener is attached.
+     * @param rootViewId The ID of the root view in the activity's layout.
+     */
+    @RequiresApi(api = Build.VERSION_CODES.VANILLA_ICE_CREAM)
+    public static void attachWindowInsetsListener(AppCompatActivity activity, int rootViewId) {
```

Comment:
couple questions:

1. Why not set the padding directly in xml layout instead ?
2. Is there no backward compatibility for edge to edge for pre Android 15 devices ?

Reply:
> Is there no backward compatibility for edge to edge for pre Android 15 devices ?

We can enable it for older versions too, but I believe it's better to focus on Android 15 and make sure that everything is fine.

Reply:
> Like fixed padding?
Nvm, I didn't realise the values are getting eval'ed at runtime. 

> We can enable it for older versions too, but I believe it's better to focus on Android 15 and make sure that everything is fine.

I think it's critical for us to be consistent across devices and we would like to make sure that if we support edge to edge, we support it on all devices and not just Android 15.

Reply:
Sorry,  Rethinking on backward compatibility, I do see value in only supporting it to Android 15 only as we do expect OS versions to impact UX in general.

Reply:
yes